dump/src/reader/compat/v1_to_v2.rs

```diff
@@ -0,0 +1,414 @@
+use std::{collections::BTreeSet, str::FromStr};
+
+use crate::reader::{v1, v2, Document};
+
+use super::v2_to_v3::CompatV2ToV3;
+use crate::Result;
+
+pub struct CompatV1ToV2 {
+    pub from: v1::V1Reader,
+}
+
+impl CompatV1ToV2 {
+    pub fn new(v1: v1::V1Reader) -> Self {
+        Self { from: v1 }
+    }
+
+    pub fn to_v3(self) -> CompatV2ToV3 {
+        CompatV2ToV3::Compat(self)
+    }
+
+    pub fn version(&self) -> crate::Version {
+        self.from.version()
+    }
+
+    pub fn date(&self) -> Option<time::OffsetDateTime> {
+        self.from.date()
+    }
+
+    pub fn index_uuid(&self) -> Vec<v2::meta::IndexUuid> {
+        self.from
+            .index_uuid()
+            .into_iter()
+            .enumerate()
+            // we use the index of the index 😬 as UUID for the index, so that we can link the v2::Task to their index
+            .map(|(index, index_uuid)| v2::meta::IndexUuid {
+                uid: index_uuid.uid,
+                uuid: uuid::Uuid::from_u128(index as u128),
+            })
+            .collect()
```

Comment:
If the indexes end up not being ordered (which I think is not the case, see my other comment).
We should probably sort them here.

Reply:
the indexes are definitely ordered, see https://github.com/meilisearch/meilisearch/pull/3170#discussion_r1037066016

Reply:
Nice thanks!